File: server/Dtos/CompraOrden.cs
namespace server.Dtos
{
    public class CompraOrdenDTO
    {
        public required int  IdProveedor { get; set; }
        public required int IdMoneda { get; set; }
        public required string RefProveedor { get; set; }
        public required string CodOrden { get; set; }
        public required int  IdEmpresa { get; set; }
        public required int EstadoCompra { get; set; }
        public required int Prioridad { get; set; }
        public required string Nota { get; set; }
        public required int IdUsrComprador { get; set; }
        public required int MontoSinImpuesto { get; set; }
        public required int MontoImpuesto { get; set; }
        public required int MontoTotal { get; set; }
        public required DateOnly FechaLimitePedido { get; set; }
        public required DateOnly FechaConfirmacion { get; set; }
        public required DateOnly FechaEntregaPlanifi { get; set; }
        public required List<CompraOrdenDetalleDTO> CompraOrdenDetalle { get; set; }
        
    }
}

File: web/src/types/res/CompraOrden.ts
import { ContactoInfoRes } from "./ContactoRes";
import { EmpresaRes } from "./EmpresaRes";
import { MonedaRes } from "./MonedaRes";
import { ProductoRes } from "./ProductoRes";
import UnidadMedidaRes from "./UnidadMedida";

export interface CompraOrdenRes {
  id: number;
  idProveedor: number;
  idMoneda: number;
  moneda: MonedaRes;
  refProveedor: string;
  codOrden: string;
  idEmpresa: number;
  estadoCompra: number;
  prioridad: number;
  nota: string;
  idUsrComprador: number;
  montoSinImpuesto: number;
  montoImpuesto: number;
  montoTotal: number;
  fechaLimitePedido: Date;
  fechaConfirmacion: Date;
  fechaEntregaPlanifi: Date;
  proveedor: ContactoInfoRes;
  empresa: EmpresaRes;
  compraOrdenDetalle: CompraOrdenDetalleRes[];
  nombre: string;
}
export interface CompraOrdenDetalleRes {
  idProducto: number;
  idUnidadMedida: number;
  idMoneda: number;
  idEmpaquetado: number;
  nombre: string;
  cantidad: number;
  precioUnitario: number;
  precioSubtotal: number;
  precioTotal: number;
  precioImpuesto: number;
  cantidadSolicitada: number;
  cantidadRecibida: number;
  cantidadPaquete: number;
  fechaEsperada: Date;
}
export interface CompraOrdenDetalleInfoRes {
  idProducto: number;
  producto:ProductoRes;
  idUnidadMedida: number;
  idMoneda: number;
  moneda: MonedaRes;
  unidadMedida: UnidadMedidaRes;
  idEmpaquetado: number;
  nombre: string;
  cantidad: number;
  precioUnitario: number;
  precioSubtotal: number;
  precioTotal: number;
  precioImpuesto: number;
  cantidadSolicitada: number;
  cantidadRecibida: number;
  cantidadPaquete: number;
  fechaEsperada: Date;
}
